File: ex1_add1/src/SimpleClosure.fs
let add (x: int) =
    printfn "Creating closure add(%d)" x // Estää kääntäjää optimoimasta sulkeumaa kokonaan pois.
    fun (y: int) -> x + y

[<EntryPoint>]
let main argv =
    printfn "%d" ((add 5) 4)
    let add10 = add 10
    printfn "%d" (add10 90)
    printfn "%d" (add10 999)
    0

File: ex1_add1/src/simple_closure.clj
(ns simple-closure
  (:gen-class))

(defn add [x]
  (fn [y]
    (+ x y)))

(defn -main []
  (println ((add 5) 4))
  (def add10 (add 10))
  (println (add10 90))
  (println (add10 999)))

